core/domain/exp_services.py

```diff
@@ -651,20 +649,37 @@ def _create_exploration(
         init_state_name=exploration.init_state_name,
         states={
             state_name: state.to_dict()
-            for (state_name, state) in exploration.states.items()},
+            for (state_name, state) in exploration.states.items()
+        },
         param_specs=exploration.param_specs_dict,
         param_changes=exploration.param_change_dicts,
         auto_tts_enabled=exploration.auto_tts_enabled,
-        correctness_feedback_enabled=exploration.correctness_feedback_enabled
-    )
-    commit_cmds_dict = [commit_cmd.to_dict() for commit_cmd in commit_cmds]
-    model.commit(committer_id, commit_message, commit_cmds_dict)
+        correctness_feedback_enabled=exploration.correctness_feedback_enabled)
+
     exploration.version += 1
+    change_list_dict = [change.to_dict() for change in change_list]
 
-    # Trigger statistics model creation.
-    exploration_stats = stats_services.get_stats_for_new_exploration(
+    new_exp_stats = stats_services.get_stats_for_new_exploration(
         exploration.id, exploration.version, exploration.states)
-    stats_services.create_stats_model(exploration_stats)
+
+    def _update_storage_models():
+        """Groups all storage calls into a function call so that they can be
+        performed as an atomic transaction.
+        """
+        # Rights to the exploration must exist before creating the exploration.
+        rights_manager.create_new_exploration_rights(
+            exploration.id, committer_id)
+
+        exploration_model.commit(committer_id, commit_message, change_list_dict)
+
+        stats_services.create_stats_model(new_exp_stats)
+
+        stats_services.create_exp_issues_for_new_exploration(
+            exploration.id, exploration.version)
+
+    transaction_services.run_in_transaction(_update_storage_models)
+
+    create_exploration_summary(exploration.id, committer_id)
```

Comment:
Probably should leave a warning note saying "don't run this in transaction because XXX"